### PR TITLE
Add `warg publish {grant,revoke}` subcommands

### PR DESCRIPTION
**Pull request by lann**

### CODE REVIEW

File: src/commands/publish.rs [hunk 1]
```diff
@@ -308,6 +319,168 @@ impl PublishYankCommand {
     }
 }
 
+/// Publish a package to a warg registry.
```
Comment (lann) [inline src/commands/publish.rs:322]: There is a lot of duplication in these subcommands that I'd like to clean up...

File: crates/protocol/src/package/model.rs [hunk 2]
```diff
@@ -55,13 +55,13 @@ impl fmt::Display for Permission {
...
 impl FromStr for Permission {
-    type Err = ();
+    type Err = String;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
             "release" => Ok(Permission::Release),
             "yank" => Ok(Permission::Yank),
-            _ => Err(()),
+            _ => Err(format!("invalid permission {s:?}")),
```
Comment (lann) [inline crates/protocol/src/package/model.rs:64]: This change allows `clap` to magically parse these as cli args.

File: crates/client/src/api.rs [hunk 3]
```diff
@@ -89,13 +89,23 @@ async fn deserialize<T: DeserializeOwned>(response: Response) -> Result<T, Clien
...
+                    "Unexpected response body: {}",
+                    String::from_utf8_lossy(&bytes)
+                );
+                ClientError::UnexpectedResponse {
                     status,
                     message: format!("failed to deserialize JSON response: {e}"),
-                }),
-            }
+                }
+            })
```
Comment (lann) [inline crates/client/src/api.rs:108]: Unrelated debug logging helped me debug my own silly problem, but it seemed worth keeping so I lazily left it in this PR.

**Comment by calvinrp** [comment]
Optionally, could we add to the README documenting these new commands?

**Review by calvinrp — APPROVED** [review]

**Pull request merged**